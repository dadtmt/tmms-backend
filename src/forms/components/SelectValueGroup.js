import React, { PropTypes } from 'react'
import { ControlLabel, FormGroup, FormControl } from 'react-bootstrap'

const SelectValueGroup = ({
  controlId,
  input,
  label,
  values
}) => (
  <FormGroup controlId={controlId}>
    <ControlLabel>{label}</ControlLabel>
    <FormControl componentClass='select' {...input}>
      {values.map(
        ({ sLabel, value }) => <option
        key={value}
        value={value}
        >
          {sLabel}
        </option>
      )}
    </FormControl>
  </FormGroup>
)

SelectValueGroup.propTypes = {
  controlId: PropTypes.string.isRequired,
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  values: PropTypes.array.isRequired
}

export default SelectValueGroup
