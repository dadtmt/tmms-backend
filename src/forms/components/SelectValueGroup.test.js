import React from 'react'
import renderer from 'react-test-renderer'

import SelectValueGroup from './SelectValueGroup'

it('should render correctly', () => {
  const props = {
    controlId: 'SelectValue',
    input: {
      value: 'default'
    },
    label: 'Select a value',
    values: [
      {
        sLabel: 'Default value',
        value: 'default'
      },
      {
        sLabel: 'Some value',
        value: 'some'
      }
    ]
  }
  expect(
    renderer.create(<SelectValueGroup {...props} />).toJSON()
  ).toMatchSnapshot()
})
