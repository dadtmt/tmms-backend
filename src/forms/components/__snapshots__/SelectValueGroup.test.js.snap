exports[`test should render correctly 1`] = `
<div
  className="form-group">
  <label
    className="control-label"
    htmlFor="SelectValue">
    Select a value
  </label>
  <select
    className="form-control"
    id="SelectValue"
    type={undefined}
    value="default">
    <option
      value="default">
      Default value
    </option>
    <option
      value="some">
      Some value
    </option>
  </select>
</div>
`;
